### PR TITLE
GraphQL improvements. 

### PR DESCRIPTION
**Pull request by crspeller**
## Summary
Improvements to the developer experience around GraphQL and other related improvements. 

Allow for inlining of GraphQL queries for more flexibility, see `webapp/src/components/rhs/rhs_main.tsx` for a good example.

Add fragment support. See `PlaybookModalFieldsFragment` for a good example. (other examples need better type handling TODO)

### CODE REVIEW

**Comment by trilopin** [comment]
@crspeller nice job here. I have some questions to fully understand this move.
- will the result of graphql(`xxx`) still be automatically typed? 
- for changes on these inline graphql queries, is `npm run graphql` still needed?
- What's the rationale behind the MetricXXX changes?

**Comment by crspeller** [comment]
> will the result of graphql(xxx) still be automatically typed?

Yes it will. 

> for changes on these inline graphql queries, is npm run graphql still needed?

Yes they will. But it now supports a watch mode that I hope to integrate with our existing watch tools

> What's the rationale behind the MetricXXX changes?

The option we where using before to use the existing type is unsupported.

**Comment by crspeller** [comment]
More changes incoming but I realized that this is good to go on its own.

**Comment by trilopin** [comment]
I want to be sure about the benefits of these changes.

> Allow for inlining of GraphQL queries for more flexibility, see webapp/src/components/rhs/rhs_main.tsx for a good example.

I felt that writing dedicated queries per view (in contrast to store them in a common file that could drive us to instinctively generalize/reuse queries) is what you mean by flexibility. Is that correct? any other benefits?

**Comment by crspeller** [comment]
@trilopin Inlining queries to bring the query closer to the user. Also the ability to use fragments like you see [here](https://github.com/mattermost/mattermost-plugin-playbooks/pull/1711/files#diff-9d271d34ee4cd7d02207425f525c79b55c162c759d821d9158fc2a743971001cR36)

**Review by trilopin — APPROVED** [review]
Thanks @crspeller 

Now I see the potential for this kind of move 🎉

File: .circleci/config.yml [hunk 1]
```diff
@@ -207,7 +207,7 @@ jobs:
       - run:
           name: Checking diff of webapp graphql generated files
           command: |
-            git --no-pager diff --exit-code webapp/src/graphql/generated_types.ts || (echo "Please run \"make graphql\" and commit the changes in webapp/src/graphql/generated_types.ts." && exit 1)
+            git --no-pager diff --exit-code webapp/src/graphql/generated/ || (echo "Please run \"make graphql\" and commit the changes in webapp/src/graphql/generated_types.ts." && exit 1)
```
Comment (iomodo) [inline .circleci/config.yml:210]:
```suggestion
            git --no-pager diff --exit-code webapp/src/graphql/generated/ || (echo "Please run \"make graphql\" and commit the changes in webapp/src/graphql/generated/." && exit 1)
```

File: webapp/src/components/playbooks_selector.tsx [hunk 2]
```diff
@@ -33,9 +33,49 @@ interface Props {
     onSelectPlaybook: (playbookId: string) => void;
 }
 
+const PlaybookModalFieldsFragment = graphql(/* GraphQL */`
```
Comment (iomodo) [inline webapp/src/components/playbooks_selector.tsx:36]: Just for me to understand we are adopting style to move GraphQL queries from `*.graphql` files to where they belong correct?

**Review by iomodo — APPROVED** [review]
Great job @crspeller !

File: webapp/src/types/playbook.ts [hunk 3]
```diff
@@ -68,11 +68,9 @@ export interface PlaybookWithChecklist extends Playbook {
...
 }
 
-export enum MetricType {
-    Duration = 'metric_duration',
-    Currency = 'metric_currency',
-    Integer = 'metric_integer',
-}
+import {MetricType} from 'src/graphql/generated/graphql';
+
+export {MetricType};
```
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/types/playbook.ts:73]: It's just a temporary thing yes. Much less disruptive.

File: webapp/src/components/sidebar/playbooks_sidebar.tsx [hunk 4]
```diff
@@ -84,7 +110,7 @@ const useLHSData = (teamID: string) => {
                     isFavorite={run.isFavorite}
                     ownerUserId={run.ownerUserID}
                     participantIDs={run.participantIDs}
-                    followerIDs={run.metadata.followers}
+                    followerIDs={run.followers}
```
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/components/sidebar/playbooks_sidebar.tsx:113]: Yup!

File: webapp/src/components/sidebar/playbooks_sidebar.tsx [hunk 5]
```diff
@@ -20,9 +22,33 @@ import {ItemContainer, StyledNavLink} from './item';
 export const RunsCategoryName = 'runsCategory';
 export const PlaybooksCategoryName = 'playbooksCategory';
 
+export const playbookLHSQueryDocument = graphql(/* GraphQL */`
```
[the remark this answers is not in the document]
Comment (crspeller) [reply webapp/src/components/sidebar/playbooks_sidebar.tsx:25]: The only thing that appears to do is tell editors that the following is a GraphQL query so they can syntax highlight it. Not a fan but it worked in my environment 🤷

File: webapp/src/components/playbooks_selector.tsx (hunk 2 above)
Comment (crspeller) [reply webapp/src/components/playbooks_selector.tsx:36]: That's correct. There may still be use cases for some shared queries, but most of the time queries will be for a single component.

**Pull request merged**